main.cpp

```diff
@@ -117,25 +120,28 @@ struct MyUpdater
 {
     static float updateLogic(U* that, float* newValue )        //10
     {
-        if (newValue == nullptr)
+        if (newValue == nullptr && newValue == nullptr)
         {
             std::cout<<"Detected Nullpointer!" << std::endl;
-            return 0.0f; 
         }
 
-        std::cout << "U's uSetPoint value: " << that->uSetPoint << std::endl;
-        that->uSetPoint = *newValue; //Acess the uSetPoint float that is part of the Pointer to a Type U* named that
-        std::cout << "U's uSetPoint updated value: " << that->uSetPoint << std::endl;
-        while( std::abs(that->uProcessVariable - that->uSetPoint) > 0.001f )
-        {    
-            /*
-             write something that makes the distance between that->uProcessVariable and that->uSetPoint get smaller
-             */
-            that->uProcessVariable += 0.5f ;
-            std::cout << that->uProcessVariable << std::endl;
+        if (that != nullptr && newValue != nullptr)
```

Comment:
if you fix the previous `if()`, you won't need this check.
- [ ] revise

Reply:
removed the second if statement and added a return 0.0f to the first if statement in case a nullpointer has been detected